File: 프로그래머스/unrated/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.js
function solution(arr, flag) {
    const answer = [];
    
    flag.forEach((el, i) => {
        const times = el ? arr[i] * 2 : arr[i];
        
        for (let j = 0; j < times; j++) {
            if (el) {
                answer.push(arr[i]);
            } else {
                answer.pop();
            }
        }
    });
    
    return answer;
}

File: 프로그래머스/unrated/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.java
import java.util.Stack;

class Solution {
    public int[] solution(int[] arr, boolean[] flag) {
        Stack<Integer> result = new Stack<>();
        
        for (int i = 0; i < flag.length; i++) {
            boolean canAdd = flag[i];
            
            if (canAdd) {
                for (int j = 0; j < arr[i]; j++) {
                    result.push(arr[i]);
                    result.push(arr[i]);
                }
            } else {
                for (int j = 0; j < arr[i]; j++) {
                    result.pop();
                }
            }
        }
        
        return result.stream().mapToInt(i -> i).toArray();
    }
}